File: 21560 - Sai Suraj/week 10/62_Unique Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp,dp_prev = [0]*n,[0]*n
        for r in range(m):
            for c in range(n):
                if r==0 or c==0:
                    dp[c]=1
                else:
                    dp[c] = dp[c-1]+dp_prev[c]
            dp,dp_prev = dp_prev,dp
        return dp_prev[n-1]

File: 21557 - Mukesh J/Week10/62.-Unique-Paths/62.-Unique-Paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] path= new int[m][n];
        int i,j;
        for(i=0;i<m;i++)
        {
            path[i][0]=1;
        }
        for(j=0;j<n;j++)
        {
            path[0][j]=1;
        }
        for(i=1;i<m;i++)
        {
            for(j=1;j<n;j++)
            {
                path[i][j]=path[i][j-1]+path[i-1][j];
            }
        }
        return path[m-1][n-1];
    }
}
